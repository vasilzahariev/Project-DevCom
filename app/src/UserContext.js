import React from 'react';

const UserContext = React.createContext({
    user: null,
    login: (userObj) => {},
    logout: () => {}
});

export default UserContext;
