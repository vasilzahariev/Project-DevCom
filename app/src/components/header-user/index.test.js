import React from 'react';
import renderer from 'react-test-renderer'
import { BrowserRouter, Route } from 'react-router-dom';
import UserContext from '../../contexts/UserContext';
import HeaderUser from '.';

describe('HeaderUser', () => {
    it('should render Admin', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <Route>
                    <UserContext.Provider value={{ user: { loggedIn: true, isAdmin: true } }}>
                        <HeaderUser />
                    </UserContext.Provider>
                </Route>
            </BrowserRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
    it('should render Admin 2', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <Route>
                    <UserContext.Provider value={{ user: { loggedIn: true, isOwner: true } }}>
                        <HeaderUser />
                    </UserContext.Provider>
                </Route>
            </BrowserRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
    it(`shouldn't render Admin`, () => {
        const tree = renderer.create(
            <BrowserRouter>
                <Route>
                    <UserContext.Provider value={{ user: { loggedIn: true } }}>
                        <HeaderUser />
                    </UserContext.Provider>
                </Route>
            </BrowserRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})
