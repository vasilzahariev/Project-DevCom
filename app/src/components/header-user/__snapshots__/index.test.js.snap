// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderUser should render Admin 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <a
      className="badge"
      href="/chat"
      onClick={[Function]}
    >
      <span
        className="MuiBadge-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          style={
            Object {
              "fontSize": 25,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H6l-2 2V4h16v12z"
          />
        </svg>
        <span
          className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary MuiBadge-invisible"
        >
          0
        </span>
      </span>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div>
      <button
        aria-haspopup="true"
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiAvatar-fallback"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`HeaderUser should render Admin 2 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <a
      className="badge"
      href="/chat"
      onClick={[Function]}
    >
      <span
        className="MuiBadge-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          style={
            Object {
              "fontSize": 25,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H6l-2 2V4h16v12z"
          />
        </svg>
        <span
          className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary MuiBadge-invisible"
        >
          0
        </span>
      </span>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div>
      <button
        aria-haspopup="true"
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiAvatar-fallback"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`HeaderUser shouldn't render Admin 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <a
      className="badge"
      href="/chat"
      onClick={[Function]}
    >
      <span
        className="MuiBadge-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          style={
            Object {
              "fontSize": 25,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H6l-2 2V4h16v12z"
          />
        </svg>
        <span
          className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary MuiBadge-invisible"
        >
          0
        </span>
      </span>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div>
      <button
        aria-haspopup="true"
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiAvatar-fallback"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </span>
      </button>
    </div>
  </div>
</div>
`;
